src/main/kotlin/com/criptext/mail/push/data/PushAPIRequestHandler.kt

```diff
@@ -55,6 +66,71 @@ class PushAPIRequestHandler(private val ctx: Context, private val manager: Notif
         }
     }
 
+    fun openEmail(metadataKey: Long, notificationId: Int, db: AppDatabase){
```

Comment:
This shouldn't have access to the whole `AppDatabase` object, only the relevant DAOs like the other methods.

Reply:
also needs tests